pkg/asset/installconfig/platformpermscheck.go

```diff
@@ -56,12 +56,17 @@ func (a *PlatformPermsCheck) Generate(dependencies asset.Parents) error {
 	case aws.Name:
 		permissionGroups := []awsconfig.PermissionGroup{awsconfig.PermissionCreateBase}
 		usingExistingVPC := len(ic.Config.AWS.Subnets) != 0
+		usingKMS := len(ic.Config.AWS.DefaultMachinePlatform.EC2RootVolume.KMSKeyARN) != 0
 		usingExistingPrivateZone := len(ic.Config.AWS.HostedZone) != 0
 
 		if !usingExistingVPC {
 			permissionGroups = append(permissionGroups, awsconfig.PermissionCreateNetworking)
 		}
 
+		if usingKMS {
+			permissionGroups = append(permissionGroups, awsconfig.PermissionKMSEncryptionKeys)
```

Comment:
Adding here will only check when the credentials mode is in default or for AWS that is mint mode. Do we also need to check passthrough mode ?

Reply:
@patrickdillon ^^

Reply:
That's a good point, but for the scope of this bug this is fine. Just make sure to mention to QE that the credential validation check is only triggered when using the default mode.

Reply:
> Adding here will only check when the credentials mode is in default or for AWS that is mint mode. Do we also need to check passthrough mode ?

@barbacbd , looking at https://github.com/openshift/installer/pull/7428/files#diff-8e36b5854a90a47985a78aa607f0ac89636aae3231ec21e55b690470e151739eL49-L51, it appears that the credentialsMode should be "".  `ic.Config.CredentialsMode` should be populated with its default value by now, correct?

Reply:
Nit: @pawanpinjarkar can you add this new code for determining value of usingKMS here: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformpermscheck.go#L68. Better to avoid interspersing with existing code for other permissionGroups.

Reply:
Yes it would be populated with the default. In AWS the default ("") is the same as Mint Mode. That is fine, but I am curious if these checks also _should_ be run when using pass through. Manual mode being skipped seems fine.

Reply:
The other issue would be that this check wouldn't be hit when the user explicitly adds `credentialsMode: Mint` to the install config

Reply:
> The other issue would be that this check wouldn't be hit when the user explicitly adds `credentialsMode: Mint` to the install config

From the description here: https://github.com/openshift/installer/blob/master/data/data/install.openshift.io_installconfigs.yaml#L1973, it appears, that is the intended behavior.

Reply:
@sadasu, @patrickdillon and I discussed this already. There is actually an enhancement that "describes" this feature. Neither of us were 100% on it. I think this is worth discussing but that is up to you. If you believe that this is good then push it for merge.

Reply:
@barbacbd I think we can go ahead with this given the above discussions( will need LGTM label) and someone from the installer team can prioritize the enhancement when feasible. Currently, I do not know what that enhancement is and when it is planned for.

Reply:
> @sadasu, @patrickdillon and I discussed this already. There is actually an enhancement that "describes" this feature. Neither of us were 100% on it. I think this is worth discussing but that is up to you. If you believe that this is good then push it for merge.

The enhancement referred to here is https://github.com/openshift/enhancements/blob/master/enhancements/installer/aws-permissions-check-bypass.md. It proposes adding a config object to allow modifying the runtime behavior of CCO. This fix takes care of the problem the enhancement is looking to solve without the ability to modify runtime behavior on day-2. So, there is value in merging this now.